File: project/suchan-onboarding/domain/src/main/java/com/survey/domain/surveyResponse/repository/SurveyResponseRepository.java
package com.survey.domain.surveyResponse.repository;

import com.survey.domain.surveyResponse.SurveyResponse;

import java.util.List;

public interface SurveyResponseRepository {
    void save(SurveyResponse surveyResponse);

    List<SurveyResponse> findBySurveyIdFetchJoin(Long surveyId);
}

File: project/sangkyeong-onboarding/domain/src/main/kotlin/org/survey/domain/response/repository/SurveyResponseRepository.kt
package org.survey.domain.response.repository

import org.survey.domain.response.model.SurveyResponse

interface SurveyResponseRepository {
    fun save(surveyResponse: SurveyResponse): Long

    fun findById(id: Long): SurveyResponse?

    fun findAllBySurveyId(surveyId: Long): List<SurveyResponse>
}
